### PR TITLE
IOS SDK documentation 

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

Provide contributor guide for creating new services for our SDK.
Document will be linked from aerogear documentation when we will have top level contribution guide for service that will include creating APB and also link to android and IOS SDK changes required for service. This will be engineering type docs as opposed to Mobile Developer docs - we going to target community or internal developers who would like to extend Aerogear Services platform.

Having these docs close to the iOS Core SDK repo would be fine e.g. in the main README or some other md file in that repo.
It could eventually end up on aerogear.org under some Contributing area, but that is out of scope for this task.

## What was done

- [x] Extensive end user documentation (generic)
- [x] SDK service template
- [x] Script for creating new SDK basing on template (it allows to create up to 64 SDK's per second ;) )
- [x] Documentation for creating new SDK
- [x] Diagram and introduction to architecture
- [x] Short description of the core module 
This will be extended, but since module is still under review I did not want to add more specific details.

## Bonus 

Documentation for release process with scripts.

##  Targets

- [x] Document how service, core is structured
- [x] What is used to build, publish
- [x] How to test changes
- [x] Provide documentation for example.

## Note 

We are using core-parser branch as base. Once core-parser will be merged this branch will change base to master.

### CODE REVIEW

**Comment by danielpassos** [comment]
@wtrocki I'm a little confused here, https://github.com/aerogear/aerogear-ios-sdk/pull/8 was from your repo this one is from upstream (AeroGear) but point to the core-parser.

I have rebased this branch `ios-doc` after merge https://github.com/aerogear/aerogear-ios-sdk/pull/8 but wanna you double check if I have missed something

**Comment by wtrocki** [comment]
@danielpassos I needed to point to upstream in order to hide commits that wereny merged to master. Since original PR is now on master I have changed base and it seems that is up to date and fully mergable with master. Thanks for sorting that out.

File: docs/getting-started.adoc [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+== Prerequisites
+
+To get started with the Mobile.Next() IOS SDK, you can set up the SDK in your existing or new project. SDK is written in Swift language. You can also run the `./example` project to see how the SDK works.
+
+To use the AeroGear SDK for iOS, you will need the following installed on your development machine:
+
+* iOS 9 or later
+* Xcode 9 or later
+* [CocoaPods](https://cocoapods.org/)
```
Comment (danielpassos) [inline docs/getting-started.adoc]: Link in markdown instead of in asciidoc

File: docs/getting-started.adoc [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+=== CocoaPods
+
+1) The AeroGear Mobile SDK for iOS is available through [CocoaPods](http://cocoapods.org)
```
Comment (danielpassos) [inline docs/getting-started.adoc]: Link in markdown instead of in asciidoc

**Comment by wtrocki** [comment]
ping @ccfoley

**Pull request merged**